src/components/tl/ucp/tl_ucp_coll.c

```diff
@@ -234,45 +234,101 @@ ucc_status_t ucc_tl_ucp_memmap_segment(ucc_tl_ucp_task_t *task,
 ucc_status_t ucc_tl_ucp_coll_dynamic_segment_init(ucc_coll_args_t   *coll_args,
                                                   ucc_tl_ucp_task_t *task)
 {
-    ucc_tl_ucp_team_t    *tl_team = UCC_TL_UCP_TASK_TEAM(task);
-    ucc_tl_ucp_context_t *ctx     = UCC_TL_UCP_TEAM_CTX(tl_team);
-    int                   i       = 0;
-    ucc_status_t          status;
+    ucc_tl_ucp_team_t    *tl_team    = UCC_TL_UCP_TASK_TEAM(task);
+    ucc_tl_ucp_context_t *ctx        = UCC_TL_UCP_TEAM_CTX(tl_team);
+    int                   i          = 0;
+    uint64_t              need_map   = 0x7;
```

Comment:
It would be preferable to use the generic interface for masks, for more readability. It requires defining a enum type instead of hardcoding `1`, `2`, `4`, e.g.,
```
typedef enum {
    UCC_TL_UCP_DYN_SEG_UPDATE_SRC    = UCC_BIT(0),
    UCC_TL_UCP_DYN_SEG_UPDATE_DST    = UCC_BIT(1),
    UCC_TL_UCP_DYN_SEG_UPDATE_GLOBAL = UCC_BIT(2),
} ucc_tl_ucp_dynamic_segment_update_mask_t;
```
, and to express `0x7` as `UCC_TL_UCP_DYN_SEG_UPDATE_SRC | UCC_TL_UCP_DYN_SEG_UPDATE_DST | UCC_TL_UCP_DYN_SEG_UPDATE_GLOBAL`
see, for example:
https://github.com/openucx/ucc/blob/985947a521d7c00e6866f032b4db4c18e5bf9e6a/src/ucc/api/ucc_def.h#L124

Reply:
done. thanks!